src/serde.rs

```diff
@@ -85,7 +82,7 @@ mod tests {
         let cid =
             Cid::try_from("bafkreibme22gw2h7y2h7tg2fhqotaqjucnbc24deqo72b6mkl2egezxhvy").unwrap();
         let bytes = serde_json::to_string(&cid).unwrap();
-        let cid2 = serde_json::from_str(&bytes).unwrap();
-        assert_eq!(cid, cid2);
+        serde_json::from_str::<Cid>(&bytes)
```

Comment:
I don't understand why we *don't* want to support this. To me it makes sense that if I have a CID encoded as bytes and I explicitly ask those bytes to be decoded as a CID, I surely want a CID back. Isn't this the whole point of Serde?

I see this outside of the context of any IPLD work. Let's say I want to use CIDs, but don't want to use IPLD. I'd like CIDs to be encoded with Serde (as it's convenient), but I'd also like to be able to decode them again.